x/market/types/types.go

```diff
@@ -164,3 +270,57 @@ var LeaseStateMap = map[string]LeaseState{
 func (obj Lease) ID() LeaseID {
 	return obj.LeaseID
 }
+
+// String implements the Stringer interface for a Lease object.
+func (obj Lease) String() string {
+	out, _ := yaml.Marshal(obj)
+	return string(out)
+}
+
+// Leases is a collection of Lease
+type Leases []Lease
+
+// String implements the Stringer interface for a Leases object.
+func (l Leases) String() string {
+	var out string
+	for _, order := range l {
+		out += order.String() + "\n"
+	}
+
+	return strings.TrimSpace(out)
+}
+
+// Accept returns whether lease filters valid or not
+func (filters LeaseFilters) Accept(obj Lease) bool {
```

Comment:
OT: It is reasonable to put these here, but they are pretty independent and I'd like to move them out of `types` at some point.